pkg/reconciler/kubernetes/adapter.go

```diff
@@ -235,6 +247,12 @@ func (g *kubeClientAdapter) Delete(ctx context.Context, manifest, namespace stri
 	for i := len(unstructs) - 1; i >= 0; i-- {
 		unstruct := unstructs[i]
 
+		// execute the delete request only if the resource exists
+		_, err := g.kubeClient.Get(unstruct.GetKind(), unstruct.GetName(), namespace)
+		if k8serr.IsNotFound(err) {
+			continue
+		}
```

Comment:
In my opinion this is not needed, because we already have `if !k8serr.IsNotFound(err)` after calling `DeleteResourceByKindAndNameAndNamespace`.

Reply:
Actually the `if !k8serr.IsNotFound(err)` is being called after the DELETE request. The main idea of this added if was to use the "cheap" get call from the cache and ensure the resource actually exists before executing the "not cheap" Delete call

Reply:
> use the "cheap" get call from the cache

Where do you see that get is using a cache?

Reply:
Sorry, my mistake, there is no cache operation there, so only get operation before delete